includes/event-date.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Wporg\TranslationEvents;
+
+use DateTime;
+use DateTimeImmutable;
+use DateTimeZone;
+use Exception;
+
+abstract class Event_Date extends DateTimeImmutable {
+	abstract public function is_end_date(): bool;
+	public $timezone;
```

Comment:
I think it would be best to make this private to prevent it from being modified from outside this class. Currently it's only being called in tests, so we could add a getter that the test would call.